src/resources/2019/DiabolicPizza.ts

```diff
@@ -0,0 +1,100 @@
+import { autosellPrice, getWorkshed, toInt, visitUrl } from "kolmafia";
+import { get as getModifier } from "../../modifier";
+import { have as haveItem } from "../../lib";
+import { $item, $monster } from "../../template-string";
+import { clamp } from "../../utils";
+
+/**
+ * @returns true if the Diabolic pizza cube is installed in workshed.
+ */
+export function installed(): boolean {
+  return getWorkshed() === $item`diabolic pizza cube`;
+}
+
+/**
+ * @returns true if the Diabolic pizza cube is installed in workshed or in inventory.
+ */
+export function have(): boolean {
+  return installed() || haveItem($item`diabolic pizza cube`);
+}
+
+/**
+ * @param item The item to be checked for validity.
+ * @returns true if an item is a valid pizza ingredient.
+ */
+export function validIngredient(item: Item): boolean {
```

Comment:
I think that this should have a function signature of ...item, and get rid of the "ingredients" version.